ledger/internal/eval.go

```diff
@@ -1171,14 +1141,15 @@ func (eval *BlockEvaluator) applyTransaction(tx transactions.Transaction, balanc
 	case protocol.ApplicationCallTx:
 		err = apply.ApplicationCall(tx.ApplicationCallTxnFields, tx.Header, balances, &ad, gi, evalParams, ctr)
 
-	case protocol.CompactCertTx:
-		// in case of a CompactCertTx transaction, we want to "apply" it only in validate or generate mode. This will deviate the cow's CompactCertNext depending of
-		// whether we're in validate/generate mode or not, however - given that this variable in only being used in these modes, it would be safe.
+	case protocol.StateProofTx:
+		// in case of a StateProofTx transaction, we want to "apply" it only in validate or generate mode. This will deviate the cow's StateProofNext depending on
+		// whether we're in validate/generate mode or not, however - given that this variable is only being used in these modes, it would be safe.
 		// The reason for making this into an exception is that during initialization time, the accounts update is "converting" the recent 320 blocks into deltas to
-		// be stored in memory. These deltas don't care about the compact certificate, and so we can improve the node load time. Additionally, it save us from
+		// be stored in memory. These deltas don't care about the state proofs, and so we can improve the node load time. Additionally, it save us from
 		// performing the validation during catchup, which is another performance boost.
 		if eval.validate || eval.generate {
-			err = balances.compactCert(tx.CertRound, tx.CertType, tx.Cert, tx.Header.FirstValid, eval.validate)
+			stateProofApplier := apply.StateProofs(cow)
```

Comment:
similarly, this line is unnecessary — just pass cow (or its old name, balances) to apply.StateProof, why have three names for the same value in the same function — cow, balances, stateProofApplier?